### PR TITLE
[Collab] Document persistence, ws authentication middleware

### PR DESCRIPTION
**Pull request by jeffrey-jian**
## PR Description

This PR implements the functionality to persist the collaborative code document into the Supabase `collaboration` table. The document shared between two collaborators in a collaboration room will be fetched on load, and updated every `x` seconds.

## Changes

**Frontend:**
1. Moved the collaboration page to `/collab/[id]`, where `id` serves as the document name and identifier for the collaboration room.

**Backend:**

- **`collaboration-gateway`:**
   1. Implemented functionality in the Hocuspocus server to fetch and save the document to Supabase.
   2. Added middleware for the WebSocket connection to ensure the below 3 criteria are met. If unfulfilled, connection will be closed.
      - The user is authenticated.
      - The collaboration entry exists in Supabase.
      - The user is one of the collaborators in the specified collaboration entry.
   3. Added logging for better traceability.
  
- **`collaboration-repository` and `collaboration-supabase`:**
   1. Added methods to support the above functionalities.

### CODE REVIEW

File: project/apps/collaboration-service/src/adapters/db/collaboration.supabase.ts [hunk 1]
```diff
@@ -83,4 +85,63 @@ export class CollaborationSupabase implements CollaborationRepository {
...
+      .maybeSingle();
+
+    if (error) {
+      throw error;
+    }
+
+    return !!data;
+  }
+
+  async fetchDocumentById(id: string): Promise<any> {
```
[the remark this answers is not in the document]
Comment (jeffrey-jian) [reply project/apps/collaboration-service/src/adapters/db/collaboration.supabase.ts:104]: thx for spotting, will add this in.

File: project/apps/collaboration-service/src/collaboration.gateway.ts [hunk 2]
```diff
@@ -1,22 +1,121 @@
...
-  }
+  constructor(
+    @Inject('AUTH_SERVICE') private readonly authServiceClient: ClientProxy,
+    private readonly collaborationRepository: CollaborationRepository,
+  ) {}
 
-  private readonly hocuspocusServer = new Hocuspocus({
+  private readonly hocuspocusServer = HocuspocusServer.configure({
     port: 1234,
+    timeout: 1000 * 60 * 60,
```
[the remark this answers is not in the document]
Comment (jeffrey-jian) [reply project/apps/collaboration-service/src/collaboration.gateway.ts]: amended.

**Review by firwer — APPROVED** [review]
lgtm

**Review by gongg21 — APPROVED** [review]
lgtm 👍

**Pull request merged**